spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/context/LoadBalancer.java

```diff
@@ -0,0 +1,50 @@
+package org.springframework.cloud.client.loadbalancer.impl.context;
+
+/**
+ * @author Spencer Gibb
+ */
+public interface LoadBalancer<T> {
+	enum Status {
+		SUCCESSS,  // Request was handled successfully
+		FAILED, // Request reached the server but failed due to timeout or internal error
+		DISCARD, // Request did not go off box and should not be counted for statistics
+	}
+
+	//TODO: add metrics
+	class CompletionContext {
+		final Status status;
+
+		public CompletionContext(Status status) {
```

Comment:
should this include a nullable throwable (ex a normal bug could abend and might be helpful)